src/engine/authentication.md

```diff
@@ -0,0 +1,52 @@
+# Authentication
+
+The `engine` JSON-RPC interface, exposed by EL and consumed by CL, needs to be authenticated. The authentication scheme chosen for thus purpose is [JWT](https://jwt.io/). 
+
+The type of attacks that this authentication scheme attempts to protect against are the following: 
+
+- RPC port exposed towards the internet, allowing attackers to exchange messages with EL engine api. 
+- RPC port exposed towards the browser, allowing malicious webpages to submit messages to the EL engine api. 
+
+The authentication scheme is _not_ designed to 
+
+- Prevent attackers with capability to read ('sniff') network traffic from reading the traffic, 
+- Prevent attackers with capability to read ('sniff') network traffic from performing replay-attacks of earlier messages. 
```

Comment:
Doesn't `iat` claim in the way it's used by this spec put a restriction on replay attacks?

Reply:
No, I wouldn't say that. There's a `10`-second window for replay attacks, which is an eternity in the case that we have an active attacker with network sniffing possibilities. 
If we actually wanted to prevent replay attacks, we'd have to enforce some sequantial `nonce` in the payloads,

Reply:
It gives us protection against "slow" replay attacks, where the user e.g. has pasted a traffic example onto a github issue, or stuff like that.